### ISSUE TITLE
[feature] adding (nodestage, nodepublish) secrets via the command line

### ISSUE THREAD

**Issue by f1-outsourcing** [ceph/ceph-csi#1809]
I am not sure if this is going to be helpful to many, I think it would be nice to have during testing. 

Currently I have problems submitting secrets via the mesos,marathon gui[3]. I keep getting that the secrets that I am sending are not found[2]. I am using their newly released mvp csi[1] (I have been trying to get info on this from mesosphere, but until now no response.)

I wonder where this also goes wrong. With the csc command line tool I am perfectly able get a block device staged (still having issues with mounting the xfs). But I think this tool is old, so it maybe sending credentials as a default, and not specific to staging or publishing (if the csi spec even offers this difference)

So maybe to get around 'temporary issues' it would be nice to specify some credentials and parse them as environment variables or so, to the plugin. The plugin sets these as a default, and overwrites them when secrets are being parsed via grpc.

[2]
rpc error: code = InvalidArgument desc = stage secrets cannot be nil or empty

[1]
https://github.com/apache/mesos/blob/master/docs/isolators/csi-volume.md#example

[3]
https://docs.d2iq.com/mesosphere/dcos/2.2/storage/external-storage/csi/

**Comment by Madhu-1** [comment]
I don't think we can implement this feature. The RPC request should carry the secrets as part of the request https://github.com/container-storage-interface/spec/blob/master/spec.md#nodestagevolume. 
cephcsi is implemented as per the CSI spec, its CO's responsibility to send the request in the RPC calls.

**Comment by f1-outsourcing** [comment]
That is why I suggested "The plugin sets these as a default, and overwrites them when secrets are being parsed via grpc.".

This would even allow the plugin to be used as stand-alone/external, similar to how docker volume drivers are being implemented. I tested rexray a bit, that has credentials stored like this and it is still a default in DCOS (I think). So I do not think such a design is bad nor conflicting with csi standards.
